### DISCUSSION TITLE
Is it possible to put HTML tags into Airflow task log. Relative HTML link in task logs

### DISCUSSION THREAD

**Discussion by PetroPliuta** [apache/airflow#26265] [Q&A]
Hello  
I would appreciate if somebody give me a piece of advice  

My problem is - I need to paste link of a specific DAG run into task log  
This is simple when I know Airflow URL (hostname) and it is permanent. I simply do this:  

```python
def my_operator(dag_run):
    airflow_url="https://example.com"
    dag_id="my_another_dag"
    from urllib.parse import quote
    run_id=quote(dag_run.run_id)
    print(f"{airflow_url}/dags/{dag_id}/grid?dag_run_id={runid}")
```

But, in my case, Airflow URL (hostname) is variable.  
So, I need another solution  
I tried to do this:  

```python
print(f"""<a href="/dags/{dag_id}/grid?dag_run_id={runid}">my run_id</a>""")
```

But Airflow does not convert HTML attributes. The string is printed "as is"    
![image_2022-09-09_135851162](https://user-images.githubusercontent.com/42343532/189335455-bfd0d2ad-fbd3-4d36-a4b0-8b63ad5c6b2b.png)

The question is:  
Is it possible to put HTML tags into log?  
Or maybe there is another solution which allows putting **relative** HTML links into log (?)

**Comment by potiuk** [comment]
You should not use links in logs. You should use operator extra links: https://airflow.apache.org/docs/apache-airflow/stable/howto/define_extra_link.html and those would likely give you what you want. Log "links" are only something that might or might not be rendered by your log browser and you shoudl not rely on the logs to provide you a clickable log (especially if it is variable depending on where the log is viewed).

**Comment by yc0** [comment]
> 
according to your question, the answer is Yes, and Airflow supports the hyperlink in the task log view since `2.1.x`.
However, it cannot be interpreted into HTML format in grid view. You should go to the exact log view, and the URL would be like this:
```
http://localhost:8080/log?dag_id=<dag_id>&task_id=<task_id>&execution_date=<time>
```
![image](https://github.com/apache/airflow/assets/10542832/09c90acb-c930-4cff-a8b3-f6dc0e0340e9)

or you can tap `see more` button to get the log view within grid view
![image](https://github.com/apache/airflow/assets/10542832/41a94f21-3405-431d-a3f2-3b773e1dd16e)

**Reply by rajrao** [reply]
Do you know why links are not rendered as links in the grid view, but do render as links when you view them via "See More"?
